coreos-installer

```diff
@@ -662,31 +685,150 @@ log() {
+    boot_partition+=( $first_track $(dc -e "${boot_partition[1]} $block_per_tracks 1 - + $block_per_tracks / p") )
+    root_partition+=($(( ${boot_partition[2]} + ${boot_partition[3]} )) "last")
+    cat > "/tmp/fdasd_conf" <<- EOF
+[$first_track,$(( ${root_partition[2]} - 1 )),native]
+[${root_partition[2]},${root_partition[3]},native]
+EOF
+
+    # format the ECKD DASD using fdasd program
+    fdasd --silent --config /tmp/fdasd_conf "${DEST_DEV}"
+
+    # copy the content of each partition
+    set -- ${boot_partition[@]} ${root_partition[@]}
+    while [ $# -gt 0 ]; do
+        dd bs=4096 if="${imagefile}" iflag=fullblock of="${DEST_DEV}" \
+            status=progress \
+            skip="$1" \
+            count="$2" \
+            seek="$(( $3 * $block_per_tracks ))"
+        if [[ $? -ne 0 ]]; then
+            log "failed to write image to ECKD DASD device"
+            exit 1
+        fi
+        shift 4
+    done
+}
+#########################################################
+#Handle s390x zFCP SCSI devices
+#########################################################
+write_image_to_zfcp_disk() {
+    log "Extracting disk image"
+    mount_tmpfs $(( $($DECOMPRESSION_TOOL ${imagefile_compressed} | wc --bytes) / 1024 / 1024 + ${TMPFS_MBSIZE} + 100 ))
+    $DECOMPRESSION_TOOL ${imagefile_compressed} > ${imagefile}
+
+    # we have to store the strings to a file because any change in this
+    # while-do block would not take effect outside
+    (while IFS='' read -r line || [ -n "$line" ]
+    do
+        if [[ $line =~ "start=" ]]
+        then
+            # we save the start= and size= here, then dd the partitions later
+            printf "%s:%s\n" \
+                "$(echo $line | cut -d: -f 2 | cut -d, -f 1 | tr -d ' ')" \
+                "$(echo $line | cut -d: -f 2 | cut -d, -f 2 | tr -d ' ')" \
+                >> /tmp/sfdisk_info
+        elif ! [[ $line =~ "label:" || $line =~ "label-id:" || $line =~ "unit:" ]]
+        then
+            continue
+        fi
+        echo $line
+    done <<< $(sfdisk -d ${imagefile})
+    ) | sfdisk "${DEST_DEV}"
+
+    while IFS='' read -r line || [ -n "$line" ]
+    do
+        eval ${line%%:*} ${line##*:}
+        dd if=${imagefile} bs=512 iflag=fullblock \
+            of="${DEST_DEV}" status=progress skip=$start seek=$start count=$size
+        if [[ $? -ne 0 ]]; then
+            log "failed to write image to zFCP SCSI disk"
+            exit 1
+        fi
+    done < /tmp/sfdisk_info
+}
 #########################################################
 #And Write the image to disk
 #########################################################
 write_image_to_disk() {
     log "Writing disk image"
 
     set -o pipefail
-    # Check for gzip and xz correspondingly
-    if [ "$(dd count=2 bs=1 if=/mnt/dl/imagefile.compressed status=none)" = "$(printf '\x1f\x8b')" ]
+    if [[ "$arch" == "aarch64" || "$arch" == "ppc64le" || "$arch" == "x86_64" ]]
```

Comment:
I'd suggest reversing this OR catching when we have an unknown arch.

Examples:

```bash
if [[ "$arch" == "s390x" ]]; then
...
else
...
fi
```

or 

```bash
if [[ "$arch" == "aarch64" || "$arch" == "ppc64le" || "$arch" == "x86_64" ]]; then
...
elif [[ "$arch" == "s390x" ]]; then
...
else
...
fi
```